api/test/unit/utilities/lottery-date-validator.spec.ts

```diff
@@ -0,0 +1,172 @@
+        constraints: [],
+        targetName: '',
+      }),
+    ).toBeFalsy();
+  });
+
+  it('should return true if reviewOrderType is lottery and lotteryOptIn is yes and listingEvents has publicLottery without startDate', () => {
+    const lotteryDateParamValidator = new LotteryDateParamValidator();
+    expect(
+      lotteryDateParamValidator.validate(
+        [
+          {
+            type: ListingEventsTypeEnum.publicLottery,
+            startDate: null,
+            startTime: new Date(),
+            endTime: new Date(),
+            id: null,
+            createdAt: null,
+            updatedAt: null,
+          },
+        ],
+        {
+          property: 'listingEvents',
+          object: {
+            reviewOrderType: ReviewOrderTypeEnum.lottery,
+            lotteryOptIn: true,
+          },
+          value: undefined,
+          constraints: [],
+          targetName: '',
+        },
+      ),
+    ).toBeFalsy();
+  });
+
+  it('should return true if reviewOrderType is lottery and lotteryOptIn is yes and listingEvents has publicLottery without startTime', () => {
+    const lotteryDateParamValidator = new LotteryDateParamValidator();
+    expect(
+      lotteryDateParamValidator.validate(
+        [
+          {
+            type: ListingEventsTypeEnum.publicLottery,
+            startDate: new Date(),
+            startTime: null,
+            endTime: new Date(),
+            id: null,
+            createdAt: null,
+            updatedAt: null,
+          },
+        ],
+        {
+          property: 'listingEvents',
+          object: {
+            reviewOrderType: ReviewOrderTypeEnum.lottery,
+            lotteryOptIn: true,
+          },
+          value: undefined,
+          constraints: [],
+          targetName: '',
+        },
+      ),
+    ).toBeFalsy();
+  });
+
+  it('should return true if reviewOrderType is lottery and lotteryOptIn is yes and listingEvents has publicLottery without endTime', () => {
+    const lotteryDateParamValidator = new LotteryDateParamValidator();
+    expect(
+      lotteryDateParamValidator.validate(
+        [
+          {
+            type: ListingEventsTypeEnum.publicLottery,
+            startDate: new Date(),
+            startTime: new Date(),
+            endTime: null,
+            id: null,
+            createdAt: null,
+            updatedAt: null,
+          },
+        ],
+        {
+          property: 'listingEvents',
+          object: {
+            reviewOrderType: ReviewOrderTypeEnum.lottery,
+            lotteryOptIn: true,
+          },
+          value: undefined,
+          constraints: [],
+          targetName: '',
+        },
+      ),
+    ).toBeFalsy();
+  });
+
+  it('should return true if reviewOrderType is lottery and lotteryOptIn is yes and listingEvents has full publicLottery event', () => {
+    const lotteryDateParamValidator = new LotteryDateParamValidator();
+    expect(
+      lotteryDateParamValidator.validate(
+        [
+          {
```

Comment:
As additional coverage could you add an additional listing event so the test has to "find" the correct event to do the validation?